.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,38 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```

Comment:
Since this job builds and pushes image to dockerhub...

```yaml
on:
  pull_request:
    branches: [ "main" ]
```

Does this mean image is pushed on creating PR?

Makes more sense to push image only on merging to main.

Reply:
ok! keep only `pull_request`

Reply:
I think you meant keep only `push`